Added a error summary table to the UG (#178)
AY2122S2-CS2103T-T13-3/tp: docs/UserGuide.md

| **Show help**                  | `help`                                                                                                                                                                                                   |
| **Clear all data**             | `clear`                                                                                                                                                                                                  |
| **Exit**                       | `exit`                                                                                                                                                                                                   |

> ## Error summary

Reviewer: ```suggestion
## Interpreting errors

The error messages you may encounter while using LinkyTime should be mostly self-explanatory. This sub-section aims to clear up any possible ambiguity related to invalid arguments and/or parameters.
```

I think having a short "intro" to this section would be good. Feel free to edit the sentence if the wording is a bit off

| Error                                                                                          | What it means                                                                                                                                                                                                                                  |
|------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| **Unknown Command**                                                                            | An unrecognised command word was supplied.{::nomarkdown}<br/> The command word was given in upper case.{:/}                                                                                                                                    |

> | **Invalid Command**                                                                            | You are not following the exact command format of the command word that was entered.{::nomarkdown} Missing some compulsory parameters. <br> The correct format would be listed after this error message for convenience.{:/}                   |

Reviewer: ```suggestion
| **Invalid Command**                                                                            | The command format entered is incorrect.{::nomarkdown}<br> Missing one or more compulsory parameters. {:/}                   |
```

| **The meeting index provided is invalid**                                                      | The index provided is out of the range of the list of meetings.                                                                                                                                                                                |
| **The module index provided is invalid**                                                       | The index provided is out of the range of the list of modules.                                                                                                                                                                                 |

> | **Names should only contain alphanumeric characters and spaces, and it should not be blank**   | Self explanatory.                                                                                                                                                                                                                              |

Reviewer: ```suggestion
| **Names should only contain alphanumeric characters and spaces, and it should not be blank**   | Check that the name is entered after the `n/` prefix.                                                                                                                                                                                                                              |
```

> | **URLs should be a valid link, and it should not be blank**                                    | The URL provided should be in one of the following formats (either `https` or `http` is fine){::nomarkdown}: <ul><li>`https://www.example.com`</li> <li>`https://example.com`</li> <li>`www.example.com`</li> <li>`example.com`</li> </ul>{:/} |

Reviewer: ```suggestion
| **URLs should be a valid link, and it should not be blank**                                    | The URL provided should be in one of the following formats: {::nomarkdown}<ul><li>`https://www.example.com`</li> <li>`http://example.com`</li><li>`www.example.com`</li><li>`example.com`</li></ul>{:/} |
```

> | **DateTime should be formatted as dd-MM-yyyy HHmm; e.g. 30-04-2022 1400**                      | The dateTime given is invalid. {::nomarkdown} <br/> <ul><li>Giving dates that do not exist 29th of February</li> <li>Giving time that does not exist like 2500 hours.</li></ul>  {:/}                                                          |

Reviewer: ```suggestion
| **DateTime should be formatted as dd-MM-yyyy HHmm; e.g. 30-04-2022 1400**                      | The dateTime given is invalid. {::nomarkdown} <br/><ul><li>Giving a date that does not exist, e.g. 30th of February.</li><li>Giving a time that does not exist, e.g. `2500`.</li></ul>  {:/}                                                          |
```

| **Meeting recurrence status can only be a single letter `Y` or `N`**                           | Self explanatory.                                                                                                                                                                                                                              |
| **Tags names should be alphanumeric**                                                          | The tags given were not alphanumeric, spaces are not allowed.                                                                                                                                                                                  |

> | **At least one field to edit must be provided.**                                               | No fields to edit were supplied together with the edit command.{::nomarkdown}<br/> As for `medit`, since there is currently only 1 field, it is compulsory as stated in the features section in this User Guide.{:/}                           |

Reviewer: ```suggestion
| **At least one field to edit must be provided.**                                               | {::nomarkdown}No fields were supplied together with the `edit`/`medit` command.{:/}                           |
```
